### PR TITLE
Add CSI snapshot API types into default restore priorities

### LINKED ISSUE #1565: Update defaultRestorePriorities for CSI resources
When restoring a CSI volume, the following resources should be restored before PVCs, in order to properly hook up references:

* volumesnapshotcontents.snapshot.storage.k8s.io
* volumesnapshot.snapshot.storage.k8s.io
* volumesnapshoclass.snapshot.storage.k8s.io 

A more complete list looks like this:

```go
var defaultRestorePriorities = []string{
    "namespaces",
    "storageclasses",
    "customresourcedefinitions",
    "volumesnapshotclass.snapshot.storage.k8s.io",
    "volumesnapshotcontents.snapshot.storage.k8s.io",
    "volumesnapshots.snapshot.storage.k8s.io",
    "persistentvolumes",
    "persistentvolumeclaims",
    "secrets",
    "configmaps",
    "serviceaccounts",
    "limitranges",
    "pods",
    "replicaset",
}
```

### PR DESCRIPTION
**Pull request by ashish-amarnath**
These changes will be behind a feature flag for csi.

Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

closes #1565 

cc @skriss  @nrb

### CODE REVIEW

**Comment by skriss** [comment]
idk what @nrb thinks, but I think it'd be OK to make this addition irrespective of the feature flag to keep things simpler.

**Comment by nrb** [comment]
@ashish-amarnath Can we keep this scoped to https://github.com/vmware-tanzu/velero/issues/1565 and have individual PRs for the other work?

**Comment by ashish-amarnath** [comment]
@nrb Sure. This PR can be just for #1526 
I don't think the changes in this PR should have any undesirable side effects.
I can remove the `WIP`. LMK

**Comment by skriss** [comment]
If we're gonna make this change irrespective of the feature flag (still waiting for @nrb's second opinion but I think it's fine), then let's add the 3 new resources to the existing var.

**Comment by nrb** [comment]
Yeah, I think this change is fine to be made outside of the feature flag.

File: pkg/cmd/server/server.go [hunk 1]
```diff
@@ -458,6 +458,9 @@ var defaultRestorePriorities = []string{
 	"customresourcedefinitions",
 	"namespaces",
 	"storageclasses",
+	"volumesnapshotclass.snapshot.storage.k8s.io",
```
Comment (skriss) [inline pkg/cmd/server/server.go]: please update the field comment to explain why these resources are where they are in the list

**Comment by skriss** [comment]
@ashish-amarnath please add a changelog file as well (`changelogs/unreleased/2318-ashish-amarnath`), then ready to merge.

**Pull request merged**